src/compositemark/boxplot.ts

```diff
@@ -44,7 +45,25 @@ export const VL_ONLY_BOXPLOT_CONFIG_PROPERTY_INDEX: {
   box: ['size']
 };
 
+export function filterUnsupportedEncChannels(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT | BoxPlotDef>) {
+  const {mark: mark, encoding: encoding, ...outerSpec} = spec;
+  const {x: x, y: y, ...nonPositionEncoding} = encoding;
+  const newNonPositionEncoding = {};
+  forEach(nonPositionEncoding, (f, c) => {
+    if (supportedEncChannels.indexOf(c) > -1) {
+      newNonPositionEncoding[c] = f;
+    } else {
+      log.warn(log.message.incompatibleChannel(c, BOXPLOT));
+    }
+  });
+  newNonPositionEncoding['x'] = x;
+  newNonPositionEncoding['y'] = y;
+
+  spec.encoding = newNonPositionEncoding;
```

Comment:
1) This can cause side effect to the original `spec`.  Use `...` and return a new spec object.  
2) `newNonPositionEncoding`  is not non-position so this is a misleading variable name.